qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -1959,28 +1973,28 @@ def add_user(self,
         ----------
         username : str
             New user's username
-        password : str
-            The user's password
+        password : str, optional
+            The user's password. If None, a new password will be generated.
         permissions : list of str, optional
             The associated permissions of a user ['read', 'write', 'compute', 'queue', 'admin']
-
+        overwrite: bool, optional
+            Overwrite the user if it already exists.
         Returns
         -------
         tuple
-            Successful insert or not
+            A tuple of (success flag, password)
         """
-        valid_permissions = {'read', 'write', 'compute', 'queue', 'admin'}
 
         # Make sure permissions are valid
-        if not valid_permissions >= set(permissions):
-            raise KeyError("Permissions settings not understood: {}".format(set(permissions) - valid_permissions))
+        if not SQLAlchemySocket.VALID_PERMISSIONS >= set(permissions):
+            raise KeyError("Permissions settings not understood: {}".format(
+                set(permissions) - SQLAlchemySocket.VALID_PERMISSIONS))
 
-        return_password = False
         if password is None:
-            password = secrets.token_urlsafe(32)
-            return_password = True
+            password = SQLAlchemySocket._generate_password()
```

Comment:
Can you call these static methods via `self` and not the class name? This applies to all instances.

Reply:
Will do.